### PR TITLE
Support layers federation

### PR DESCRIPTION
**Pull request by runcom**
Based on https://github.com/containers/image/pull/147

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LTM

File: cmd/skopeo/layers.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ var layersCmd = cli.Command{
 			if !strings.HasPrefix(digest, "sha256:") {
 				digest = "sha256:" + digest
 			}
-			r, blobSize, err := rawSource.GetBlob(digest)
+			r, blobSize, err := rawSource.GetBlob(types.BlobInfo{Digest: digest})
```
Comment (mtrmac) [inline cmd/skopeo/layers.go]: This should probably set `Size` to -1 (unknown), depending on how #147 defines the `GetBlob` requirements.

**Comment by runcom** [comment]
@mtrmac PTAL

**Comment by runcom** [comment]
@mtrmac merging because there's only one tiny small changes here in skopeo to unblock PRs in containers/image (that otherwise fails)

**Comment by mtrmac** [comment]
ACK to the layers fix, but please redo without the `hack/vendor.sh` part.

**Comment by runcom** [comment]
@mtrmac I did this here https://github.com/projectatomic/skopeo/pull/257 no?

**Comment by mtrmac** [comment]
#257 cleaned up #254; this is another instance, on top of #257.

**Pull request merged**